sqlstore/src/main/resources/db/migration_beforeMigrate/05_standard.sql

```diff
@@ -419,6 +419,32 @@ CREATE PROCEDURE addTissueType(
   END IF;
 END//
 
+--  Use inconjunction with the spreadsheet formula: =CONCATENATE("CALL addIdentity('",A2,"', '",B2,"', '",C2,"', ", IF(ISBLANK(D2), "NULL", CONCATENATE("'", D2, "'")), ", ", IF(ISBLANK(E2), "NULL", CONCATENATE("DATE '", E2, "'")), ", '", F2, "', '", G2, "');")
+DROP PROCEDURE IF EXISTS addIdentity//
+CREATE PROCEDURE addIdentity(
+  iProject varchar(255),
+  iExternalName varchar(255),
+  iAlias varchar(255),
+  iDescription varchar(255),
+  iDate date,
+  iScientificName varchar(255),
+  iSampleType varchar(255)
+) BEGIN
+  DECLARE newSampleId bigint(20);
+  
+  IF NOT EXISTS (SELECT 1 FROM Identity WHERE externalName = iExternalName) THEN
```

Comment:
Needs to be project-constrained as well. There's no reason we can't have the same external name in two projects. 
This also needs to confirm that there is no pre-existing identity with the custom alias.